EWL-3987: Feature/ewl 3987 refactor articles

rpama opened the pull request:
<!-- NOTE: Please just put "N/A" for any section below that isn't applicable to the work you've done, do not omit entirely. -->

## Ticket(s)

**Jira Ticket**

- [EWL-3987: Refactor Topics Article Preview Related Patterns to use Pseudo-Patterns ](https://issues.ama-assn.org/browse/EWL-3987)


## Description

Removes previous article preview patterns and replaces with one template and json for several pseudo-patterns. Updates templates that contained this patterns to use the pseudo-patterns correctly. Makes some adjustments to styles for our vertical spacing issues, but this is kind of hinky.


## To Test

- [x] Molecules > Components > Article Preview (and variants)
- [x] Use pattern lab's show pattern info to track the other patterns that use these^^
- [x] See that nothing looks terribly broken
- [x] Review the css changes for any obvious badness


## Relevant Screenshots/GIFs

Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.


## Remaining Tasks

- [ ] ~Need Robert to confirm with UX the element order for the video variant pattern.~
Since the direction was a change from what we had before, I've requested a separate ticket for that work from @froboy.

## Additional Notes

Anything more to add?

labbydev commented on styleguide/source/_patterns/02-organisms/06-topic/01-topic-hero/_topic-hero.scss:
```diff
@@ -15,13 +15,12 @@
+//   @include breakpoint($bp-med) {
+//     @include child-top-gutters($gutter / 2);
+//   }
+// }
```
This should be cleaned out if it isn't needed anymore

labbydev commented on styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/_topic-topic-articles.scss:
```diff
@@ -19,9 +19,9 @@
 .topic_topic-articles .topic_article-preview > * {
-  @include gutter($margin-top-half...);
+  //@include gutter($margin-top-half...);
 }
```
This should be cleaned out

labbydev commented on styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/topic-topic-articles.json:
```diff
@@ -0,0 +1,18 @@
+      "video": true
+    }
+  }
+}
```
Line ending

labbydev commented on styleguide/source/_patterns/02-organisms/06-topic/05-topic-topic-articles/topic-topic-articles.twig:
```diff
@@ -1,4 +1,4 @@
-    {% include 'molecules-article-preview-video' with { 'class': default }  %}
+    {% include "molecules-article-preview" with {'topic_article' : topicTopicArticle.image } %}
+    {% include "molecules-article-preview" with {'topic_article' : topicTopicArticle.video } %}
 </div>
```
Line ending

labbydev commented on styleguide/source/assets/css/style.scss:
```diff
@@ -209,7 +209,7 @@ So feel free to use any of these approaches. Or don't. It's totally up to you.
 @import "../../_patterns/01-molecules/08-lists/14-list-topic-related-articles/list-topic-related-articles";
 @import "../../_patterns/02-organisms/06-topic/08-topic-related-articles/topic-related-articles";
-@import "../../_patterns/01-molecules/06-components/22-article-preview/02-article-preview-related/article-preview-related";
+//@import "../../_patterns/01-molecules/06-components/22-article-preview/02-article-preview-related/article-preview-related";
```
Cleanup

labbydev commented on styleguide/source/_patterns/02-organisms/06-topic/01-topic-hero/topic-hero.json:
```diff
@@ -0,0 +1,18 @@
+      "video": true
+    }
+  }
+}
```
Line ending

labbydev commented on styleguide/source/_patterns/01-molecules/08-lists/14-list-topic-related-articles/list-topic-related-articles.json:
```diff
@@ -2,13 +2,18 @@
     }
   }
-}
+}
```
Line ending

labbydev commented on styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~video.json:
```diff
@@ -0,0 +1,10 @@
+    "title": "Article Video Title",
+    "video": true
+  }
+}
```
Line ending

labbydev commented on styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~related.json:
```diff
@@ -0,0 +1,9 @@
+    "metadata": "",
+    "title": "Related Article Title"
+  }
+}
```
Line ending

labbydev commented on styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview~related-no-image.json:
```diff
@@ -0,0 +1,9 @@
+    "metadata": "",
+    "title": "Related Article Title"
+  }
+}
```
Line ending

labbydev commented on styleguide/source/_patterns/01-molecules/06-components/22-article-preview/_article-preview.scss:
```diff
@@ -1,25 +1,64 @@
+      padding-right: 0;
+    }
+  }
+}
```
Line ending

rpama commented:
I have removed the commented out code. I was being overly cautious in the PR and then forgot to do clean-up. Whoops!

I have not, however, addressed the line-ending comments since I'm not entirely sure how we want to proceed with this. I'd like to get this PR merged (after @chelsiejohnston takes one more look for anything wonky with respect to the topics layouts) and then address the line-endings holistically. I've created spike ticket [EWL-4500](https://issues.ama-assn.org/browse/EWL-4005) for tackling the line ending question.

chelsiejohnston commented on styleguide/source/_patterns/01-molecules/06-components/22-article-preview/article-preview.twig:
```diff
@@ -1,9 +1,29 @@
+        <div class="topic_article-preview_description">
+          {% include "atoms-paragraph" only %}
+        </div>
+        {% include "molecules-article-metadata" with { 'class': 'topic_article-preview_metadata' } %}
```
Suggestion to move this include outside of this conditional - this fixes the videos not having metadata and is a truer reflection of the content model

rpama commented:
Updated pattern info in the `article-preview.md` file, but ticket EWL-3793 doesn't exist in Jira anymore so that was omitted.

rpama commented:
@chelsiejohnston I believe I've now addressed all three of the issues you noted. The template's a little inelegant in its conditionals, but some of of that will go away once we get the ticket saying we don't need the weird title-position-swap for the video variant.

chelsiejohnston commented:
LGTM, hooray! Thanks @rpama 

I know we were planning on having a conversation about this PR before we merged it, so I'm going to leave it here for now.
